v4/export/writer_util.go

```diff
@@ -189,6 +190,13 @@ func WriteInsert(pCtx context.Context, tblIR TableDataIR, w storage.Writer, file
 			}
 			counter += 1
 			wp.AddFileSize(uint64(bf.Len()-lastBfSize) + 2) // 2 is for ",\n" and ";\n"
+			failpoint.Inject("ChaosBrokenMySQLConn", func(val failpoint.Value) {
+				brokenPoint := val.(int)
+				if counter >= brokenPoint {
+					tblIR.Close()
+					failpoint.Return(errors.New("connection is closed"))
+				}
```

Comment:
🤔 so now we will kill multiple connections?

Reply:
No. We will kill connection only for one time.
Setting `ChaosBrokenMySQLConn=$broken_point*return($broken_point)` is to run `brokenPoint` times, but the first `brokenPoint-1` times it won't enter into this if block.

Reply:
so why not use `1%1*return()` which will trigger at 1% probability and run at most 1 time.

Reply:
Then maybe sometimes this test can pass and sometimes this test will fail? 🤔

Reply:
You could make the percent higher (e.g. `9%1*return()`) to avoid the "run 0 times" case. Otherwise I don't see this is much different than using `$RANDOM`.

Reply:
How about directly failing one time like this?

Reply:
yeah it's fine though it's not really "chaos" now 🙃